### PR TITLE
Add constants and log utils

### LINKED ISSUE #11: Migrate `ballerina-runtime`

### PR DESCRIPTION
**Pull request by snelusha**
This pull request introduces new error handling and logging utilities, along with a set of constants for symbol flags. The main changes include adding stack trace support to errors, structured logging for critical failures, and a new constants package for commonly used values. Comprehensive unit tests have also been added to verify the new error handling features.

**Error handling and logging enhancements:**

* Added `ErrorWithStackTrace` type and the `DecorateWithStackTrace` function in `errors.go` to support attaching stack traces to errors when the `BAL_BACKTRACE` environment variable is enabled.
* Implemented `LogBadSad` and `PrintCrashLog` functions in `errors.go` for structured logging of critical, unexpected errors, including timestamp and severity level.

**Constants and symbol flags:**

* Introduced a new `constants.go` file with commonly used string constants and a set of symbol flag bitmasks, plus a utility function `IsFlagOn` to check flag values.

**Testing improvements:**

* Added `errors_test.go` with unit tests for stack trace decoration, error formatting, and edge cases for the new error handling utilities.

**Refactoring and imports:**

* Updated imports in `errors.go` to support new error handling and logging features.

Closes #11

### CODE REVIEW

File: common/constants/constants.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	OPTIONAL      int64 = 4096
+	REMOTE        int64 = 32768
+	CLIENT        int64 = 65536
+	RESOURCE      int64 = 131072
+	SERVICE       int64 = 262144
+	TRANSACTIONAL int64 = 33554432
+	CLASS         int64 = 268435456
+	ISOLATED      int64 = 536870912
+	ENUM          int64 = 8589934592
+	ANY_FUNCTION  int64 = 549755813888
```
Comment (heshanpadmasiri) [inline common/constants/constants.go]: Usually bit mask flags are defined like this https://pkg.go.dev/go/build#ImportMode. I think you still need uint64, but might be worth it to make a type alias (you can attach methods to alias directly)

File: common/errors/errors.go [hunk 2]
```diff
@@ -59,3 +66,96 @@ func NewIllegalArgumentError(argument any) *IllegalArgumentError {
...
+}
+
+func DecorateWithStackTrace(err error) error {
+	if os.Getenv("BAL_BACKTRACE") != "true" {
+		return err
+	}
+
+	if err == nil {
+		return nil
+	}
```
Comment (heshanpadmasiri) [inline common/errors/errors.go]:
```suggestion
	if err == nill || os.Getenv("BAL_BACKTRACE") != "true" {
		return err
	}
```

File: common/errors/errors.go [hunk 3]
```diff
@@ -59,3 +66,96 @@ func NewIllegalArgumentError(argument any) *IllegalArgumentError {
...
+
+	const (
+		maxDepth = 32
+		skip     = 2 // skip DecorateWithStackTrace and its caller
+	)
```
Comment (heshanpadmasiri) [inline common/errors/errors.go]: These are single use right? I think it is easier to just pass the value at the call site

File: common/errors/errors_test.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	// Error message should contain original error
+	if !strings.Contains(decorated.Error(), "test error with stack") {
+		t.Errorf("Error message should contain original error text")
+	}
+
+	// Error message should contain stack trace
+	errorMsg := decorated.Error()
+	if !strings.Contains(errorMsg, "\n\tat ") {
```
Comment (heshanpadmasiri) [inline common/errors/errors_test.go:64]: I think it is cleaner for you to check for the actual error message instead of matching part by part. It clear state what you are expecting

File: common/errors/errors.go [hunk 5]
```diff
@@ -59,3 +66,96 @@ func NewIllegalArgumentError(argument any) *IllegalArgumentError {
...
+
+	stack := make([]uintptr, maxDepth)
+	length := runtime.Callers(skip, stack[:])
+	return &errorWithStackTrace{
+		err:   err,
+		stack: stack[:length],
+	}
+}
+
+func (e *errorWithStackTrace) StackTrace() []byte {
```
Comment (heshanpadmasiri) [inline common/errors/errors.go]: Do we need this to be public?

File: common/errors/errors_test.go (hunk 4 above)
Comment (snelusha) [reply common/errors/errors_test.go:64]: Exact match would be flaky since the stack trace contains absolute paths and architecture
